hy/_compat.py

```diff
@@ -35,3 +35,24 @@ def wr_long(f, x):
 else:
     def raise_empty(t, *args):
         raise t(*args)
+
+def isidentifier(x):
+    if x in ('True', 'False', 'None', 'print'):
+        # `print` is special-cased here because Python 2's
+        # keyword.iskeyword will count it as a keyword, but we
+        # use the __future__ feature print_function, which makes
+        # it a non-keyword.
+        return True
+    if keyword.iskeyword(x):
+        return False
+    if PY3:
+        return x.isidentifier()
+    if x.rstrip() != x:
+        return False
+    import tokenize as T
+    from StringIO import StringIO
```

Comment:
You should prefer `io.StringIO`/`io.BytesIO` over the `StringIO` package

Reply:
Done.